src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,628 @@
+                $frequency = RRule::fromFrequency($frequency);
+            } elseif (strpos($frequency, 'FREQ=') !== false) {
+                $frequency = new RRule($frequency);
+            }
+
+            if ($frequency instanceof RRule) {
+                $value['interval'] = $frequency->getInterval();
+                switch ($frequency->getFrequency()) {
+                    case RRule::DAILY:
+                        if (
+                            $frequency->getInterval() <= 1
+                            && strpos($frequency->jsonSerialize(), 'INTERVAL=') === false
+                        ) {
+                            $this->setFrequency(RRule::DAILY);
+                        } else {
+                            $this
+                                ->setFrequency(static::CUSTOM_EXPR)
+                                ->setCustomFrequency(RRule::DAILY);
+                        }
+
+                        break;
+                    case RRule::WEEKLY:
+                        if (! $frequency->getByDay() || empty($frequency->getByDay())) {
+                            $this->setFrequency(RRule::WEEKLY);
+                        } else {
+                            $value['weekly-fields'] = $this->weeklyField->loadWeekDays($frequency->getByDay());
+                            $this
+                                ->setFrequency(static::CUSTOM_EXPR)
+                                ->setCustomFrequency(RRule::WEEKLY);
+                        }
+
+                        break;
+                    case RRule::MONTHLY:
+                    case RRule::YEARLY:
+                        $isMonthly = $frequency->getFrequency() === RRule::MONTHLY;
+                        if ($frequency->getByDay() || $frequency->getByMonthDay() || $frequency->getByMonth()) {
+                            $this->setFrequency(static::CUSTOM_EXPR);
+
+                            if ($isMonthly) {
+                                $value['monthly-fields'] = $this->monthlyFields->loadRRule($frequency);
+                                $this->setCustomFrequency(RRule::MONTHLY);
+                            } else {
+                                $value['annually-fields'] = $this->annuallyFields->loadRRule($frequency);
+                                $this->setCustomFrequency(RRule::YEARLY);
+                            }
+                        } elseif ($isMonthly && $frequency->getInterval() === 3) {
+                            $this->setFrequency(RRule::QUARTERLY);
+                        } else {
+                            $this->setFrequency($frequency->getFrequency());
+                        }
+
+                        break;
+                    default:
+                        $this->setFrequency($frequency->getFrequency());
+                }
+
+                $value['frequency'] = $this->getFrequency();
+                $value['custom-frequency'] = $this->getCustomFrequency();
```

Comment:
No matter what we do with `BaseFormElement::init()`, you should maybe avoid calling getters in a setter if possible.

Instead of directly calling `set..` above, you could save these to local variables and then call the setters here where you're storing them in `$value` as well.

Reply:
Even then, we still have a problem with this.
https://github.com/Icinga/ipl-web/pull/101/commits/37f653cfa4173943427b9766a1921176fdb9102e#diff-3bbb2b8560be734af1a67988cb06e5dad50794bdf9ff2e95228051201215ed0eR353